account_cost_center/models/account_invoice.py

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+# Copyright 2015 Onestein (<http://www.onestein.eu>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import models, fields, api
+from lxml import etree
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    cost_center_id = fields.Many2one(
+        'account.cost.center', string='Cost Center',
+        help="Default Cost Center")
+
+    @api.model
+    def line_get_convert(self, line, part, date):
+        res = super(AccountInvoice, self).line_get_convert(line, part, date)
+        if line.get('cost_center_id'):
+            res['cost_center_id'] = line['cost_center_id']
+        return res
+
+    def fields_view_get(self, cr, uid, view_id=None, view_type='form',
+                        context=None, toolbar=False, submenu=False):
+        res = super(AccountInvoice, self).fields_view_get(
+            cr, uid, view_id=view_id, view_type=view_type,
+            context=context, toolbar=toolbar, submenu=submenu)
+        if not context:
+            context = {}
+        if not context.get('cost_center_default', False):
+            if view_type == 'form':
+                view_obj = etree.XML(res['arch'])
+                invoice_line = view_obj.xpath("//field[@name='invoice_line']")
```

Comment:
you can replace this hack with https://github.com/OCA/server-tools/tree/8.0/base_view_inheritance_extension by now